couchbase/io/http_session.hxx

```diff
@@ -405,9 +410,10 @@ class http_session : public std::enable_shared_from_this<http_session>
 
     void do_read()
     {
-        if (stopped_) {
+        if (stopped_ || reading_ || !stream_->is_open()) {
```

Comment:
Little suspicious as this doesn't appear related to unknown bucket bootstrap crashing.

Reply:
I agree with you.